test/e2e/openshift/openshift_test.go

```diff
@@ -85,6 +85,62 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		}
 	})
 
+	It("should whitelist pods on compute nodes", func() {
+		namespaces := []string{
+			"openshift-infra",
+			"openshift-node",
+			"openshift-metrics",
+			"kube-",
+			"default",
+		}
+
+		whitelist := []string{
+			"sync-",
+			"ovs-",
+			"sdn-",
+		}
+		list, err := pod.GetAll("")
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, pod := range list.Pods {
+			skip := false
+			// we have a whitelist of namespaces that we will skip
+			for _, pattern := range namespaces {
+				if strings.Contains(pod.Metadata.Namespace, pattern) {
+					skip = true
+					break
+				}
+			}
+			if skip {
+				skip = false
+				continue
+			}
+
+			found := false
+			if strings.Contains(pod.Spec.Nodename, "compute") {
+				for _, pattern := range whitelist {
```

Comment:
think about ensuring that the intent of your test is clear to a 3rd party.  What you're testing is: "no pods should be on a compute node unless they start with a daemonsetPrefix".  Try to use patterns and variable names that make this clear, e.g. perhaps something like "isDaemonsetPod := matchPrefix()... ; assert(isDaemonsetPod)"

Reply:
This makes sense.  Will update.